code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -7,7 +7,7 @@
 /mob/living/silicon/ai/compose_track_href(atom/movable/speaker, namepart)
 	var/mob/M = speaker.GetSource()
 	if(M)
-		return "<a href='?src=\ref[src];track=[namepart]'>"
+		return "<a href='?src=\ref[src];track=[html_encode(M.name)]'>"

```

Comment:
don't switch to M.name, people will show up as 'unknown' in the radio without a voice changing mask. Use the 'namepart' argument